controllers/statefulset.go

```diff
@@ -410,7 +411,15 @@ func (r *SingleClusterReconciler) buildSTSConfigMap(
 		return fmt.Errorf("failed to build config map data: %v", err)
 	}
 
-	// Replace config map data since we are supposed to create a new config map.
+	// Replace config map data if differs since we are supposed to create a new config map.
+	if reflect.DeepEqual(confMap.Data, configMapData) {
+		r.Log.Info(
+			"Already existed configmap is same as current configmap, so no change required",
```

Comment:
This log may become very frequent.
I think we can remove this log and add the log only if configMap is getting updated.